Compute and use the screen scale factor in OSWorldRenderer

ronsaldo opened the pull request:
This PR adds the computation code for the screenScaleFactor into OSWorldRenderer. The effect of these changes is in replicating the previous behavior before the VM were not marked as DPI aware with the old renderer by drawing into a smaller resolution display and then scale it up when composing into the window.

This also enables texture filtering which is required to make scaling look good, and it fixes the methods for updating a rectangle in the OSSDL2AthensRenderer.

ronsaldo committed:
I added the usage of the screen scale factor to the OSWorldRenderer.
Use a smaller screen scale factor rounding factor.
I fixed the implement of OSDL2AthensRenderer >> #updateRectangle:

guillep reviewed (changes requested):
Hi Ronie, I believe the changes are good. My only point: can the magic numbers some little comment to say what they are and why?

There is a `0.5` in `convertWindowMouseEventPosition:` and a `0.1` that magically changed to `0.05` in `fetchDPI`.

ronsaldo committed:
Fix some comments of the reviewes.
Use rounded instead of (x + 0.5) floor. They have the same behavior with negative numbers unlike C.

ronsaldo commented:
@guillep Are these changes good enough?

guillep reviewed (approved):
Yes! Thanks!!
